docs/guides/napari_models.md

```diff
@@ -0,0 +1,222 @@
+# Napari models and events
+
+This document explains the links between the three main components of napari:
+python models, Qt classes and vispy classes, with code examples. This knowledge
+is not necessary to use napari and is more aimed at developers interested in
+understanding the inner workings of napari. This document assumes you're
+familiar with basic usage of napari.
+
+The three main components:
+
+* python models describing objects - these are able to operate without the GUI
```

Comment:
```suggestion
* python models describing components in the napari application - these are able to operate without the GUI
```

I think it might be nice to avoid "object" here to avoid confusion with programming concept.

Reply:
Thanks, this is better